frontend/src/api/auth.ts

```diff
@@ -1,13 +1,34 @@
-import axios from "axios";
+import axios, { AxiosError } from "axios";
 import { User } from "../model/User";
 
 export const login = async (username: string, password: string) => {
     try {
-        const response = await axios.post("/api/auth/login", {username,password});
+        const response = await axios.post("/api/auth/login", { username, password });
         if (response.status === 200) {
             return response.data as User;
         }
     } catch (err) {
         return null;
     }
-};
\ No newline at end of file
+};
+
+export const register = async (username: string, email: string, password: string, confirm: string) => {
+    if(password !== confirm) {
+        return { status: 400, message: "Salasanat eivät täsmää"};
+    }
+    try {
+        const response = await axios.post("/api/auth/register", { username, password, email });
+        if (response.status == 200) {
+            const user: User = response.data as User;
+            return { status: 200, data: user };
+        }
+        else {
+            return { status: 400, message: "Ei sallittu" };
+        }
+    } catch (err) {
+        const error = err as AxiosError<Error>;
+        const status = error.response?.status;
+        const msg = error.response?.data;
+        return { status: status, message: msg };
+    }
+};
```

Comment:
I have decided to return only status and message to the main program. That is because we also make another error in this code (in the beginning), if password and confirm password are not the same. 

Another alternative could be to return error in this code, in lines: 17, 26 and 34.
       17 return response.status(400).send("Salasanat eivät täsmää");
       26 return response.status(response.status).send("Ei sallittu"); 
       34 return error.response?